src/cfml/system/endpoints/S3.cfc

```diff
@@ -0,0 +1,193 @@
+/**
+*********************************************************************************
+* Copyright Since 2014 CommandBox by Ortus Solutions, Corp
+* www.coldbox.org | www.ortussolutions.com
+********************************************************************************
+* @author Brad Wood, Luis Majano, Denny Valliant
+*
+* I am the s3 endpoint.
+*/
+component accessors="true" implements="IEndpoint" singleton {
+
+    // DI
+    property name="systemSettings"  inject="SystemSettings";
+    property name="fileSystemUtil"  inject="FileSystem";
+    property name="httpsEndpoint"   inject="commandbox.system.endpoints.HTTPS";
+
+    // Properties
+    property name="namePrefixes" type="string";
+    property name="iamRolePath" type="string";
+
+    function init() {
+        setNamePrefixes('s3');
+        setIamRolePath('169.254.169.254/latest/meta-data/iam/security-credentials/');
```

Comment:
What is the IP address?  Will this ever change?

Reply:
This IP address is one made available to AWS EC2 instances in order to retrieve instance metadata: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html - it shouldn't change.

Reply:
Ok, cool.  Just checking.